ui/src/pages/Detail.tsx

```diff
@@ -70,11 +71,28 @@ const Detail = () => {
 
     const name: string = labels[MetricName]
 
+    let to: number = Date.now()
```

Comment:
If possible, `to`, `from` parsing logic and `updateTimeRange` callback can be extracted into a `useTimeRange` hook, so that the timeRange parsing and setting logic would be in one place inside that hook without getting spilled into this component. 

But I guess since this is the only page that uses timeRange, this hasn't surfaced as a problem.